src/accelerate/utils/modeling.py

```diff
@@ -143,7 +143,12 @@ def set_module_tensor_to_device(
         elif value is not None or torch.device(device) != module._parameters[tensor_name].device:
             param_cls = type(module._parameters[tensor_name])
             kwargs = module._parameters[tensor_name].__dict__
-            new_value = param_cls(new_value, requires_grad=old_value.requires_grad, **kwargs).to(device)
+            if (param_cls.__name__ == "Int8Params"):
```

Comment:
Changed to test param_cls.__name__ for Int8 parameters ("Int8Params") to only pass kwargs if true, else do not send kwargs

Reply:
Perfect, thanks!